src/connector/src/parser/unified/debezium.rs

```diff
@@ -535,9 +540,376 @@ pub fn extract_bson_id(id_type: &DataType, bson_doc: &serde_json::Value) -> Acce
+                    .decode(binary)
+                    .unwrap_or_default();
+                let bytea = ScalarImpl::Bytea(bytes.into());
+                Some(bytea)
+            } else {
+                return Err(type_error(datum));
+            }
+        }
+
+        DataType::Struct(struct_fields) => {
+            let mut datums = vec![];
+            for (field_name, field_type) in struct_fields.iter() {
+                let field_datum = extract_bson_field(field_type, datum, Some(field_name))?;
+                datums.push(field_datum);
+            }
+            let value = StructValue::new(datums);
+
+            Some(ScalarImpl::Struct(value))
+        }
+
+        DataType::List(list_type) => {
+            let Some(d_array) = datum.as_array() else {
+                return Err(type_error(datum));
+            };
+
+            let mut builder = list_type.create_array_builder(d_array.len());
+            for item in d_array {
+                builder.append(extract_bson_field(list_type, item, None)?);
+            }
+            Some(ScalarImpl::from(ListValue::new(builder.finish())))
+        }
+
+        _ => {
+            if let Some(field_name) = field {
+                unreachable!(
+                    "DebeziumMongoJsonParser::new must ensure {field_name} column datatypes."
+                )
+            } else {
+                let type_expected = type_expected.to_string();
+                unreachable!(
+                    "DebeziumMongoJsonParser::new must ensure type of `{type_expected}` matches datum `{datum}`"
+                )
+            }
+        }
+    };
+    Ok(field_datum)
+}
+
+fn bson_extract_date(bson_doc: &serde_json::Value, type_expected: &DataType) -> AccessResult {
+    // according to mongodb extended json v2
+    // the date could be:
+    //
+    // the timestamp type could be:
+    //
+    // both Canonical and Relaxed format:
+    // {"$timestamp": {"t": 1630454400, "i": 1}}
+    //
+    // Canonical: {"$date": {"$numberLong": "1630454400000"}}
+    // date is encoded as number of milliseconds since the Unix epoch
+    //
+    // Relaxed: {"$date": "2021-09-01T00:00:00.000Z"}
+    // date is encoded as ISO8601 string
+
+    let datum = &bson_doc["$date"];
+
+    let type_error = || AccessError::TypeError {
+        expected: type_expected.to_string(),
+        got: match bson_doc {
+            serde_json::Value::Null => "null",
+            serde_json::Value::Bool(_) => "bool",
+            serde_json::Value::Number(_) => "number",
+            serde_json::Value::String(_) => "string",
+            serde_json::Value::Array(_) => "array",
+            serde_json::Value::Object(_) => "object",
+        }
+        .to_owned(),
+        value: datum.to_string(),
+    };
+
+    let res = if let serde_json::Value::String(ref _iso8601_str) = datum {
+        // relaxed format
+        let Ok(parsed) = iso8601::deserialize(datum) else {
```

Comment:
There is already `chrono` or `speedate`, why `time` crate here?

Reply:
This part is used for relaxed format, won't do in this PR, I'll remove it.